Reference/conformance/conformance.pb.swift

```diff
@@ -1929,11 +1903,23 @@ public struct Conformance_TestAllTypes: ProtobufGeneratedMessage {
     get {return _storage._optionalNestedMessage ?? Conformance_TestAllTypes.NestedMessage()}
     set {_uniqueStorage()._optionalNestedMessage = newValue}
   }
+  public var hasOptionalNestedMessage: Bool {
+    return _storage._optionalNestedMessage != nil
+  }
+  public mutating func clearOptionalNestedMessage() {
+    return _storage._optionalNestedMessage = nil
+  }
 
   public var optionalForeignMessage: Conformance_ForeignMessage {
     get {return _storage._optionalForeignMessage ?? Conformance_ForeignMessage()}
     set {_uniqueStorage()._optionalForeignMessage = newValue}
   }
+  public var hasOptionalForeignMessage: Bool {
```

Comment:
Agreed, the dot-into behavior takes advantage of Swift's copy-on-write behavior in a really unique and clean way. Other languages effectively have this as well (Objective-C provides it automatically, C++ through the `mutable` methods, and so forth), but Swift's is probably the most elegant both in its user facing API and in its implementation because of the features of the language.